pkg/proto/configuration/grpc/grpc.proto

```diff
@@ -35,6 +35,10 @@ message ClientConfiguration {
   //
   // Header names must be lower case.
   repeated string forward_metadata = 4;
+
+  // Token-based authentication with Google.
+  // See https://grpc.io/docs/guides/auth/
+  ClientOAuthConfiguration credentials = 6;
```

Comment:
I guess it's best to call this 'oauth' instead of 'credentials' to make it explicit that OAuth is the mechanism that's used here.

Reply:
ok

Reply:
The comment above also mentions Google specifically though, as well as the underlying message, is it wise to call it OAuth then?

Reply:
Yeah. Something like `// OAuth token-based authentication.` ?

Reply:
That'd mean moving the Google stuff out of the underlying message though, but then this can also be changed to a oneof?

Reply:
Google authentication is also based on OAuth, right? So it's perfectly fine to keep that in ClientOAuthConfiguration.

Reply:
I was thinking about `google_default_credentials` which is now move to a oneof so yeah, this works 👍